graphics.ts

```diff
@@ -97,8 +171,159 @@ class Graphics {
+     * @param y The y coordinate for the center of the polygon.
+     * @param r The radius of the polygon.
+     * @param n The number of sides in the polygon
+     * @param v The rotation of the polygon in radians. Default is 0.
+     */
+    fillPolygon(x: number, y: number, r: number, n: number, v: number = 0) {
+        this.makePolygon(x, y, r, n, true, v);
+    }
+
+    /**
+     * Draws a polygon at the given coorinates with the radius and number of sides specified.
+     * @param x The x coordinate for the center of the polygon.
+     * @param y The y coordinate for the center of the polygon.
+     * @param r The radius of the polygon.
+     * @param n The number of sides in the polygon
+     * @param v The rotation of the polygon in radians. Default is 0.
+     */
+    drawPolygon(x: number, y: number, r: number, n: number, v: number = 0) {
+        this.makePolygon(x, y, r, n, false, v);
+    }
+
+    /**
+     * The actual implementation of the making of the polygon. Draws the polygon, and fills it if specified.
+     * @param x The x coordinate for the center of the polygon.
+     * @param y The y coordinate for the center of the polygon.
+     * @param r The radius of the polygon.
+     * @param n The number of sides in the polygon
+     * @param fill A boolean indicating if the polygon should be filled.
+     * @param v The rotation of the polygon in radians.
+     */
+    private makePolygon(x: number, y: number, r: number, n: number, fill: boolean, v: number) {
+        let ang = Math.PI * 2 / n;
+        this.ctx.translate(x , y);
+        this.ctx.rotate(v + Math.PI);
+        this.ctx.beginPath();
+        this.ctx.moveTo(Math.sin(ang) * r, Math.cos(ang) * r);
+        for (let i = 0; i <= n; i++) {
+            this.ctx.lineTo(Math.sin(ang * i) * r, Math.cos(ang * i) * r);
+        }
+        if (fill) {
+            this.ctx.fill();
+        }
+        else {
+            this.ctx.stroke();
+        }
+        this.ctx.rotate(-(v + Math.PI));
+        this.ctx.translate(-x , -y);
+    }
+
+    /**
+     * Fills a custom shape specified by array of coordinate points.
+     * @param ps The array containing tuples of coordinate points.
+     */
+    fillShape(ps: [number, number][]) {
+        this.makeShape(ps, true);
+    }
+
+    /**
+     * Draws a custom shape specified by array of coordinate points.
+     * @param ps The array containing tuples of coordinate points.
+     */
+    drawShape(ps: [number, number][]) {
+        this.makeShape(ps, false);
+    }
+
+    /**
+     * The actial implementation that makes the custom shape, and fills it if specified.
+     * @param ps The array containing tuples of coordinate points.
+     * @param fill A boolean indicating if the shape should be filled.
+     */
+    private makeShape(ps: [number, number][], fill: boolean) {
+        this.ctx.beginPath();
+        this.ctx.moveTo(ps[0][0], ps[0][1]);
+        ps.forEach(p => {

```

Comment:
for some reason it bugs me, that you use .forEach here, and for in makePolygon.. :D

Reply:
Yea, i prefer the simplicity of the forEach but it is only in makeShape i operate on an Array. I'll leave it like this for now, unless you specifically request it to match ;)